### ISSUE TITLE
Indent rule issue regarding object in multiline function parameters

### ISSUE THREAD

**Issue by jchristin** [eslint/eslint#6990]
**What version of ESLint are you using?**
3.3.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```
{
    "env": {
        "node": true
    },
    "rules": {
        "indent": ["error", "tab"]
    }
}
```

**What did you do? Please include the actual source code causing the issue.**
I linted the following code:

```
"use strict";

var React = require("react");

React.DOM.div({
        className: "row"
    },
    React.createElement("h1"),
    React.createElement("h2"),
    React.createElement("h3"),
    React.createElement("h4")
);
```

**What did you expect to happen?**
I expected to not have any lint errors.

**What actually happened? Please include the actual, raw output from ESLint.**
I got the following lint errors:

```
/home/ubuntu/workspace/test.js
  6:3  error  Expected indentation of 1 tab character but found 2   indent
  7:2  error  Expected indentation of 0 tab characters but found 1  indent

✖ 2 problems (2 errors, 0 warnings)
```

**Comment by emileber** [comment]
I'm having the same issue with collection as parameter.

```javascript
var test = new Backbone.Collection([{
        test: 2,  // line 2
    },
    {
        test: 3
    }
]);
```

If I run `eslint test-eslint.js`:

```
C:\whatever\test-eslint.js
  2:9  error  Expected indentation of 4 space characters but found 8  indent
  3:5  error  Expected indentation of 0 space characters but found 4  indent

✖ 2 problems (2 errors, 0 warnings)
```

And if I run `eslint test-eslint.js --fix`:

```javascript
var test = new Backbone.Collection([{
    test: 2,
},
    {
        test: 3
    }
]);
```

**Comment by gyandeeps** [comment]
This is why indent flags this as an issue:

For single objects it expects:

```js
var test = new Backbone.Collection({
    test: 3
});
```

For multiple objects, it expects

```js
var test = new Backbone.Collection([
    {
        test: 2,  // line 2
    },
    {
        test: 3
    }
]);
```

**Comment by platinumazure** [comment]
@gyandeeps Could we consider trying to enhance this to support the use case in the original post? I'm fairly sure that style is common.

**Comment by gyandeeps** [comment]
based on things and so many customization we already have in indent rule, it is kinda of trivial.
If anyone want to take up thing challenge i will be happy to review.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
